virttest/utils_conn.py

```diff
@@ -1941,41 +1989,68 @@ def conn_setup(self):
             pattern_to_repl = {r".*sasl_allowed_username_list\s*=.*":
                                'sasl_allowed_username_list=%s' % sasl_allowed_username_list}
             self.client_libvirtdconf.sub_else_add(pattern_to_repl)
-        else:
+        elif not self.sasl_type == 'plain':
             keytab = ""
             sasldb = "sasldb_path: /etc/libvirt/passwd.db"
-        pattern_to_repl = {r".*mech_list\s*:\s*.*":
-                           "mech_list: %s" % self.sasl_type,
-                           r".*keytab\s*:\s*.*": keytab,
-                           r".*sasldb_path\s*:\s*.*": sasldb}
+        if self.sasl_type == 'plain':
+            pattern_to_repl = {r"^mech_list\s*:\s*.*": "mech_list: %s"
+                                                       % self.sasl_type}
+        else:
+            pattern_to_repl = {r".*mech_list\s*:\s*.*": "mech_list: %s"
+                                                        % self.sasl_type,
+                               r".*keytab\s*:\s*.*": keytab,
+                               r".*sasldb_path\s*:\s*.*": sasldb}
         self.client_saslconf.sub_else_add(pattern_to_repl)
 
         # restart libvirtd service on server
         if restart_libvirtd == "yes":
             try:
+                if not self.run_on_remote:
+                    current_session = client_session
+                    runner = process.run
+                else:
+                    current_session = remote.wait_for_login('ssh',
```

Comment:
Could we use `current_session = remote_session` ? Any reason to create a separate session for current_session?

Reply:
I found, the session might be interfere, when the libvirtd services are restarted and therefore there are two independent sessions.

Reply:
@kamilvarga Sorry, server_session works on my environment, could you please explain some details about the problem you faced? The session is RemoteSession(ShellSession) object, I think it should not be affected by libvirtd service restarting. This looks weird~
And below are my testing:
```
>>> conn_obj=utils_conn.UNIXConnection(xxx)
>>> conn_obj.client_session.cmd("hostname")
'dell-per740-41.lab.eng.pek2.redhat.com\n'
>>> conn_obj.server_session.cmd("hostname")
'dell-per740-43.lab.eng.pek2.redhat.com\n'
>>> from virttest import utils_libvirtd
>>> server_libvirtd=utils_libvirtd.Libvirtd(session=conn_obj.server_session)
>>> server_libvirtd.is_running()
False
>>> server_libvirtd.restart()
True
>>> server_libvirtd.is_running()
True
>>> conn_obj.server_session.cmd("hostname")
'dell-per740-43.lab.eng.pek2.redhat.com\n'
>>> conn_obj.client_session.cmd("hostname")
'dell-per740-41.lab.eng.pek2.redhat.com\n'

```

Reply:
Will check again with full script.

Reply:
Reworked and tested with the following results:
<pre><font color="#2A7BDE">JOB ID     : 20f8df233d8e53671d985160efb74b3227b64c5b</font>
<font color="#2A7BDE">JOB LOG    : /root/avocado/job-results/job-2021-06-01T18.03-20f8df2/job.log</font>
 (01/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.default_unix_config: <font color="#33DA7A">PASS</font> (58.55 s)
 (02/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.readonly: <font color="#33DA7A">PASS</font> (57.84 s)
 (03/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.customized_unix_sock_dir: <font color="#33DA7A">PASS</font> (59.84 s)
 (04/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.default_config_and_start_vm: <font color="#33DA7A">PASS</font> (59.98 s)
 (05/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.allow_sasl_plain: <font color="#33DA7A">PASS</font> (72.84 s)
 (06/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.allow_sasl_krb_user: <font color="#33DA7A">PASS</font> (101.98 s)
 (07/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.allow_sasl_users: <font color="#33DA7A">PASS</font> (70.86 s)
 (08/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.socket_access_controls: <font color="#33DA7A">PASS</font> (70.60 s)
 (09/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.socket_with_auth_conf: <font color="#33DA7A">PASS</font> (56.44 s)
 (10/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.socket_with_polkit_and_acl_control: <font color="#33DA7A">PASS</font> (63.93 s)
 (11/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.socket_with_polkit_rw: <font color="#33DA7A">PASS</font> (63.77 s)
 (12/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.no_allowed_sasl_user: <font color="#33DA7A">PASS</font> (114.74 s)
 (13/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.allow_sasl_user_with_readonly: <font color="#33DA7A">PASS</font> (113.20 s)
 (14/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.readonly_without_auth: <font color="#33DA7A">PASS</font> (59.84 s)
 (15/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.socket_access_controls.cfg_file: <font color="#33DA7A">PASS</font> (68.87 s)
 (16/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.socket_access_controls.socket_file: <font color="#33DA7A">PASS</font> (61.93 s)
 (17/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.negative_testing.socket_with_polkit_ro: <font color="#33DA7A">PASS</font> (62.51 s)
<font color="#2A7BDE">RESULTS    : PASS 17 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0</font>
<font color="#2A7BDE">JOB TIME   : 1220.71 s</font></pre>